rust/frameworks/tw_utxo/src/modules/transaction_util.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright © 2017 Trust Wallet.
+
+use bitcoin::consensus::deserialize;
+use bitcoin::Transaction;
+use tw_coin_entry::coin_context::CoinContext;
+use tw_coin_entry::error::prelude::*;
+use tw_coin_entry::modules::transaction_util::TransactionUtil;
+use tw_encoding::hex::decode;
+
+pub struct UtxoTransactionUtil;
+
+impl TransactionUtil for UtxoTransactionUtil {
+    fn calc_tx_hash(&self, coin: &dyn CoinContext, encoded_tx: &str) -> SigningResult<String> {
+        Self::calc_tx_hash_impl(coin, encoded_tx)
+    }
+}
+
+impl UtxoTransactionUtil {
+    fn calc_tx_hash_impl(_coin: &dyn CoinContext, encoded_tx: &str) -> SigningResult<String> {
+        let tx = decode(encoded_tx).map_err(|_| SigningErrorType::Error_input_parse)?;
```

Comment:
That's not that simple as it supposes to be. `tw_utxo` crate will be used in many different Bitcoin-like chains, eg Zcash, Verge etc. The mentioned ones have different transaction structure and address types, so `rust-bitcoin` cannot be used to deserialize such transactions. I plan to implement transaction deserializer soon, and I can implement `TransactionUtil::calc_tx_hash` for Bitcoin chain as well later.

Reply:
Thanks for pointing it out. I will add some comments in the source code. I think keeping it for now isn't a problem because currently, UtxoTransactionUtil is only enabled for Bitcoin. If you support Zcash/Verge someday, you can remove this portion of the code.

Reply:
Then please move it to `tw_bitcoin` crate, it shouldn't be a part of `tw_utxo`

Reply:
Okay, I have renamed `UtxoTransactionUtil` to `BitcoinTransactionUtil` and moved it to the `tw_bitcoin` crate.